commands/deploy_file.js

```diff
@@ -37,50 +31,38 @@ exports.builder = {
 }
 
 exports.handler = async function (argv) {
-    const content = fs.readFileSync(argv.file)
-    
-    transaction = null
-    txBuilder = archethic.newTransactionBuilder("hosting")
-        .setContent(content)
+   
+    setcontent(argv.file)
     
     const address = archethic.deriveAddress(argv.seed, 0)
     
-    try {
-        index = await archethic.getTransactionIndex(address, argv.endpoint)
-    } catch (e) {
-        console.error(chalk.red(e.message))
-        return
-    }
-
-    transaction = txBuilder
-        .build(argv.seed, index)
-        .originSign(originPrivateKey)
+    index = await getindex(address, argv.endpoint)
+    
+    transaction = buildtxn(argv.seed, index)
 
 
     try {
-        const { fee: fee, rates: rates } = await archethic.getTransactionFee(transaction, argv.endpoint)
         
-        const ok = await yesno({
-            question: chalk.yellow('The transaction would cost ' +fee+ ' UCO ($ ' +rates.usd+ ' € ' +rates.eur+ '). Do you want to confirm ?')
-        });
-
+        const ok =  await feeconfirmation(transaction, argv.endpoint)
+      
         if (ok)
         {
         archethic.waitConfirmations(transaction.address, argv.endpoint, function(nbConfirmations) {
```

Comment:
This part could be wrapped into some function, as it's recurrent and obvious with need to wait confirmations.